src/utils/push.js

```diff
@@ -1,128 +1,211 @@
+            await this.displayNotification(remoteMessage);
           }
         }
-      });
-      unsubscribes.push(
-        // Specifies the action to take when the user presses the notification in the foreground.
-        notifee.onForegroundEvent(async ({type, detail}) => {
+      } catch (error) {
+        console.error(error);
+      }
+    });
+
+    return () => {
+      notifee.onForegroundEvent(async ({type, detail}) => {
+        try {
           if (type === EventType.PRESS && detail.notification && isSendbirdNotification(detail.notification.data)) {
+            sb.markPushNotificationAsClicked(detail.notification.data);
             navigationRef.current?.navigate('Notifications');
           }
-        }),
-      );
-    }
+        } catch (error) {
+          console.error(error);
+        }
+      });
+    };
+  }
+  startOnBackground() {
+    sb.markPushNotificationAsDelivered(data);
+    // Specifies the action to take when the user presses the notification in the background.
+    notifee.onBackgroundEvent(async ({type, detail}) => {
+      try {
+        if (type === EventType.PRESS && detail.notification && isSendbirdNotification(detail.notification.data)) {
+          sb.markPushNotificationAsClicked(detail.notification.data);
+          navigationRef.current?.navigate('Notifications');
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    });
 
-    if (Platform.OS === 'ios') {
-      PushNotificationIOS.addEventListener('notification', async notification => {
+    messaging().setBackgroundMessageHandler(async remoteMessage => {
+      try {
+        if (isSendbirdNotification(remoteMessage.data)) {
+          sb.markPushNotificationAsDelivered(remoteMessage.data);
+          await this.displayNotification(remoteMessage);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    });
+  }
+  displayNotification = async message => {
+    try {
+      const sendbird = parseSendbirdNotification(message.data);
+      await notifee.displayNotification({
+        id: String(sendbird.message_id),
+        title: `${sendbird['push_title'] || sendbird.channel.name || 'Message received'}`,
+        body: sendbird.message,
+        data: message.data,
+        android: {
+          channelId: this.ANDROID_NOTIFICATION_CHANNEL_ID,
+          importance: 4,
+          largeIcon: sendbird.sender?.profile_url || sendbird.channel.channel_url,
+          circularLargeIcon: true,
+          pressAction: {id: 'default'},
+          showTimestamp: true,
+          timestamp: sendbird.created_at,
+        },
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  };
+}
+class iOSNotificationHandler {
+  startOnAppOpened() {
+    PushNotificationIOS.getInitialNotification().then(async notification => {
+      try {
+        const data = notification?.getData();
+        const isClicked = data?.userInteraction === 1;
```

Comment:
This issue was discovered while implementing the latest version of Notifee.

On iOS, when a push notification is clicked, the information that the push notification was clicked is communicated through the `didReceiveNotificationResponse` delegate in the native layer (AppDelegate.mm).

Previously, it was possible to handle clicks using `react-native-push-notification-ios`, but the latest version of Notifee overrides the implementation of `didReceiveNotificationResponse`. (https://github.com/invertase/notifee/issues/925)

As a result, the push click event is not passed to the `react-native-push-notification-ios` library from `didReceiveNotificationResponse`, causing the previous code to not work.

Instead, the click event can be handled through Notifee's foreground event.

Example: https://github.com/sendbird/react-native-sample-notifications-war-room/pull/1/files#diff-16197fad13a80965a19877fb31010a9a4a011f3b406de9e9ed7f1e05e7bd9528R68-R77